src/object_file.rs

```diff
@@ -1,322 +1,28 @@
-}
-
-impl Relocation {
-    fn new(relocation: LLVMRelocationIteratorRef, object_file: LLVMObjectFileRef) -> Self {
-        assert!(!relocation.is_null());
-
-        Relocation {
-            relocation,
-            object_file,
-        }
-    }
-
-    pub fn get_offset(&self) -> u64 {
-        unsafe {
-            LLVMGetRelocationOffset(self.relocation)
-        }
-    }
-
-    pub fn get_symbols(&self) -> SymbolIterator {
-        let symbol_iterator = unsafe {
-            // REVIEW: Is this just returning a single Symbol (given the name) and not a full iterator?
-            LLVMGetRelocationSymbol(self.relocation)
-        };
-
-        SymbolIterator::new(symbol_iterator, self.object_file)
-    }
-
-    pub fn get_type(&self) -> (u64, &CStr) {
-        let type_int = unsafe {
-            LLVMGetRelocationType(self.relocation)
-        };
-        let type_name = unsafe {
-            CStr::from_ptr(LLVMGetRelocationTypeName(self.relocation))
-        };
-
-        (type_int, type_name)
-    }
-
-    pub fn get_value(&self) -> &CStr {
-        unsafe {
-            CStr::from_ptr(LLVMGetRelocationValueString(self.relocation))
-        }
-    }
-}
-
-#[derive(Debug)]
-pub struct SymbolIterator {
-    symbol_iterator: LLVMSymbolIteratorRef,
-    object_file: LLVMObjectFileRef,
-}
-
-impl SymbolIterator {
-    fn new(symbol_iterator: LLVMSymbolIteratorRef, object_file: LLVMObjectFileRef) -> Self {
-        assert!(!symbol_iterator.is_null());
-
-        SymbolIterator {
-            symbol_iterator,
-            object_file,
-        }
-    }
-}
-
-impl Iterator for SymbolIterator {
-    type Item = Symbol;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        // REVIEW: Should it compare against 1? End checking order might also be off
-        let at_end = unsafe {
-            LLVMIsSymbolIteratorAtEnd(self.object_file, self.symbol_iterator) == 1
-        };
-
-        if at_end {
-            return None;
-        }
-
-        let symbol = Symbol::new(self.symbol_iterator);
-
-        unsafe {
-            LLVMMoveToNextSymbol(self.symbol_iterator)
-        }
-
-        Some(symbol)
-    }
-}
-
-impl Drop for SymbolIterator {
-    fn drop(&mut self) {
-        unsafe {
-            LLVMDisposeSymbolIterator(self.symbol_iterator)
-        }
-    }
-}
-
-#[derive(Debug)]
-pub struct Symbol {
-    symbol: LLVMSymbolIteratorRef,
-}
-
-impl Symbol {
-    fn new(symbol: LLVMSymbolIteratorRef) -> Self {
-        assert!(!symbol.is_null());
-
-        Symbol {
-            symbol,
-        }
-    }
-
-    pub fn get_name(&self) -> &CStr {
-        unsafe {
-            CStr::from_ptr(LLVMGetSymbolName(self.symbol))
-        }
-    }
-
-    pub fn size(&self) -> u64 {
-        unsafe {
-            LLVMGetSymbolSize(self.symbol)
-        }
-    }
-
-    pub fn get_address(&self) -> u64 {
-        unsafe {
-            LLVMGetSymbolAddress(self.symbol)
-        }
-    }
-}
+use cfg_if::cfg_if;
+
+cfg_if! {
```

Comment:
We should just do
```rust
#[cfg(any(feature = "llvm3-6", feature = "llvm3-7", feature = "llvm3-8", feature = "llvm3-9", feature = "llvm4-0", feature = "llvm5-0", feature = "llvm6-0", feature = "llvm7-0", feature = "llvm8-0"))]
mod object_file;
#[llvm_versions(3.6..=8.0)]
pub use self::object_file::{Binary, BinaryRef, ObjectFile};
#[cfg(not(any(feature = "llvm3-6", feature = "llvm3-7", feature = "llvm3-8", feature = "llvm3-9", feature = "llvm4-0", feature = "llvm5-0", feature = "llvm6-0", feature = "llvm7-0", feature = "llvm8-0")))]
mod binary;
#[llvm_versions(9.0..=latest)]
pub use self::binary::{Binary, BinaryRef, BinaryType};
```

Could you explain this big file change a bit? Link to related docs maybe? Both? I don't have any context for why this is necessary